netsnmpagent.py

```diff
@@ -679,6 +692,83 @@ def DisplayString(self, initval = None, oidstr = None, writable = True, context
 			"asntype"       : ASN_OCTET_STR
 		}
 
+	# ObjectIdentifier represents SNMP OBJECT IDENTIFIERS
+	# The expected value syntax is dot notation string,
+	#   eg: ".1.3.6.1.2.1.2.2.1.1"
+	# or alternatively if MIB support is enabled the fully qualified MIB value,
+	#   eg: "IF-MIB::ifIndex"
+	def ObjectIdentifier(self, initval = None, oidstr = None, writable = False, context = "", callback = None):
+		agent = self
+
+		class ObjectIdentifier(object):
+			def __init__(self):
+				self._flags     = WATCHER_MAX_SIZE & WATCHER_SIZE_UNIT_OIDS
+				self._asntype   = ASN_OBJECT_ID
+				self._max_size  = MAX_OID_LEN * ctypes.sizeof(c_oid)
+				self._set_oid_value(initval)
+
+				if oidstr:
+					# Prepare the netsnmp_handler_registration structure.
+					self._callback_handler = None
+					if callback != None:
+						# We defined a Python function that needs a ctypes conversion so it can
+						# be called by C code such as net-snmp. That's what SNMPNodeHandler() is
+						# used for. However we also need to store the reference in "self" as it
+						# will otherwise be lost at the exit of this function so that net-snmp's
+						# attempt to call it would end in nirvana...
+						self._callback_handler = _build_callback_handler(callback)
+
+					handler_reginfo = agent._prepareRegistration(oidstr, writable)
+					handler_reginfo.contents.contextName = b(context)
+
+					# Create the netsnmp_watcher_info structure.
+					watcher = libnsX.netsnmp_create_watcher_info(
+						self.cref(),
+						self._data_size,
+						self._asntype,
+						self._flags
+					)
+					watcher._maxsize = self._max_size
+
+					# Register handler and watcher with net-snmp.
+					result = libnsX.netsnmp_register_watched_instance(
+						handler_reginfo,
+						watcher
+					)
+					if result != 0:
+						raise netsnmpAgentException("Error registering variable with net-snmp!")
+
+					if self._callback_handler is not None:
+						_inject_custom_handler(self._callback_handler, handler_reginfo)
+
+					# Finally, we keep track of all registered SNMP objects for the
+					# getRegistered() method.
+					agent._objs[context][oidstr] = self
+
+			def _set_oid_value(self, oid_value):
+				if oid_value is not None:
+					oid, oid_len = agent.determine_oid_and_length(oid_value)
```

Comment:
Does `agent.determine_oid_and_length(oid_value)` resolve and return the `.`-separated OID for a string-formatted human readable OID? Will we raise errors there if no such OID exists?

Reply:
Ah, nvm, I just read this PR in the wrong order. I see what's going on now